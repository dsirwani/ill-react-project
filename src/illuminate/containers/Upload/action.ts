export enum UPLOAD_ACTIONS {
    UPLOAD_DATA_INITIATE = 'UPLOAD_DATA_INITIATE',
    UPLOAD_DATA_ERROR = 'UPLOAD_DATA_ERROR',
    UPLOAD_DATA_SUCCESS ='UPLOAD_DATA_SUCCESS',
    RESET_UPLOAD_DATA ='RESET_UPLOAD_DATA'
  };