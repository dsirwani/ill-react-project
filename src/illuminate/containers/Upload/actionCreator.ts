import ACTIONS from '../../../utils/configureActions';
import { UploadActionTypes } from './type';

export const initUpload = (data: any): UploadActionTypes => (
	{
		type: ACTIONS.UPLOAD_DATA_INITIATE,
		payload: data
	}
);

export const uploadSuccess = (data: any): UploadActionTypes => (
	{
		type: ACTIONS.UPLOAD_DATA_SUCCESS,
		payload: data
	}
);

export const uploadError = (error: any): UploadActionTypes => (
	{
		type: ACTIONS.UPLOAD_DATA_ERROR,
		payload: error
	}
);

export const resetUpload = (): UploadActionTypes => (
	{
		type: ACTIONS.RESET_UPLOAD_DATA,
	}
);
