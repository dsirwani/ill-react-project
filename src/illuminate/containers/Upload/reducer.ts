import ACTIONS from '../../../utils/configureActions';
import { UploadState } from './type';

const initialState: UploadState = {
	loading: false,
	error: '',
	errorMessage: '',
	data: null
};

export const uploadReducer = (state: UploadState = initialState, action: any) => {
	switch (action.type) {
		case ACTIONS.UPLOAD_DATA_INITIATE:
			return ({
				...state,
				loading: true,
				error: false,
				errorMessage: '',
				data: null
			});

		case ACTIONS.UPLOAD_DATA_SUCCESS:
			return ({
				...state,
				loading: false,
				error: false,
				errorMessage: '',
				data: action?.payload
			});

		case ACTIONS.UPLOAD_DATA_ERROR:
			return ({
				...state,
				loading: false,
				error: true,
				errorMessage: action?.payload,
				data: null
			});

		case ACTIONS.RESET_UPLOAD_DATA:
			return ({
				...initialState
			});

		default:
			return { ...state };
	}
};