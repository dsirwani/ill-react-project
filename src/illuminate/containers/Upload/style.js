import { fade, makeStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  paper: {
    minHeight: 300,
    minWidth: 300,
    marginTop: 50,
    textAlign: 'center',
    // backgroundColor: theme.palette.common.white,
  },
  nav: {
    textAlign: 'left',
    // color: '#0096dc',
  },
  inputSelect: {
    // color: theme.palette.common.black,
  },
  formControl: {
    // color: theme.palette.common.black,
    paddingBottom: 20,
    minWidth: 135,
  },
  progressLoader: {
    color: '#0096dc',
  },
  fileInput: {
    width: 135,
  },
  inputOverflow: {
    display: 'block',
    overflow: 'hidden !important',
    whiteSpace: 'nowrap !important',
    textOverflow: 'ellipsis !important',
  },
  fileSelect: {
    // color: theme.palette.common.white,
  },
  inputDisplay: {
    display: 'none',
    // color: theme.palette.common.black,
  },
  themeButton: {
    color: '#000000',
    // backgroundColor: '#0096dc',

    '&:hover,&:focus': {
      // backgroundColor: '#0096dc',
    },
  },
}));