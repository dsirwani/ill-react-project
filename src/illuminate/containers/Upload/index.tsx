import React, { useState, useEffect } from 'react';
import { Link } from "react-router-dom";
import {
  Grid,
  Container,
  Paper,
  InputLabel,
  MenuItem,
  FormHelperText,
  FormControl,
  Select,
  Button,
  CircularProgress,
} from '@material-ui/core';
import { useSelector, useDispatch } from 'react-redux';
import { uploadActionCreator } from '../../../utils/configureActionCreators';
import { RootState } from '../../../utils/injectReducer';
import { UploadState as IUploadData } from './type';
import ToastNotifier from '../../components/ToastNotifier';
import ArrowBackRoundedIcon from '@material-ui/icons/ArrowBackRounded';
import { useI18n } from '../../../hooks/useI18n';
import { useStyles } from './style';

export interface UploadProps {
};

const Upload: React.FC<UploadProps> = () => {
  const classes = useStyles();
  const i18n = useI18n();
  const [file, setFile] = useState('');
  const [filename, setFilename] = useState('Choose File');
  const [fileValidations, setFileValidation] = useState('Required');
  const [workbook, setWorkbook] = React.useState(1);

  const uploadData: IUploadData = useSelector((state: RootState) => state.uploadData);
  const dispatch = useDispatch();
  const acceptVariants = ".xlsx,.xls,application/vnd.ms-excel,application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"

  useEffect((): void => {
    dispatch(uploadActionCreator.resetUpload());
  }, []);

  const handleChange = (event: any) => {
    setWorkbook(event.target.value);
  };

  const onFileChange = (event: any) => {
    setFile(event.target.files[0])
    setFilename(event.target.files[0] ? event.target.files[0].name : 'Choose File')
    setFileValidation('Required')
  }

  const checkFileType = (file: any): boolean => {
    if (!(/\.(xlsx|xls)$/i).test(file.name)) {
      setFileValidation('Files of type xlsx and xls are allowed')
      setFile('')
      setFilename('Choose File');
      return false
    }
    return true
  };

  const minFileSize = (file: any): boolean => {
    if (file.size < 5120) {
      setFileValidation('File size should be more than 5KB')
      setFile('')
      setFilename('Choose File');
      return false
    }
    return true
  };

  const onSubmit = (event: any) => {
    event.preventDefault()
    if (file) {
      if (minFileSize(file) && checkFileType(file)) {
        dispatch(uploadActionCreator.initUpload({file, workbook}));
      }
    }
    else {
      setFileValidation('No file selected');
    }
  }

  return (
    <Grid container direction="column" className={classes.root} justify="center" alignItems="center" >
      <Grid item>
        <Paper className={classes.paper}>
          <Container maxWidth="sm">
            <Grid item className={classes.nav}>
              <Link to="/" title="Dashboard"><ArrowBackRoundedIcon color="primary" className={classes.nav} fontSize="large" /></Link>
            </Grid>
            <Grid item>
              <FormControl required className={classes.formControl}>
                <InputLabel id="input-file-label" className={classes.inputSelect} >
                  {i18n._('Select File')}
                </InputLabel>
                <Select
                  className={classes.inputSelect}
                  labelId="input-file-label"
                  id="select-label"
                  value={workbook}
                  onChange={handleChange}
                >
                  <MenuItem className={classes.fileSelect} value={1}>{i18n._('Client Workbook')}</MenuItem>
                  <MenuItem className={classes.fileSelect} value={2}>{i18n._('Financial')}</MenuItem>
                  <MenuItem className={classes.fileSelect} value={3} disabled>{i18n._('Global 2000')}(G2K)</MenuItem>
                </Select>
                {/* <FormHelperText>Required</FormHelperText> */}
              </FormControl>
            </Grid>
            <Grid item>
              <FormControl className={classes.formControl}>
                <input
                  accept={acceptVariants}
                  className={classes.inputDisplay}
                  id="contained-button-file"
                  type="file"
                  onChange={onFileChange}
                />
                <label htmlFor="contained-button-file" title={i18n._(filename)} className={classes.fileInput}>
                  <Button variant="outlined" color="primary" className={classes.inputOverflow} component="div">
                    <div className={classes.inputOverflow}>
                      {i18n._(filename)}
                    </div>
                  </Button>
                </label>
                <FormHelperText>{i18n._(fileValidations)}</FormHelperText>
              </FormControl>
            </Grid>
            <Grid item>
              <FormControl className={classes.formControl}>
                {uploadData.loading ? <CircularProgress className={classes.progressLoader} />
                  :
                  <Button variant="contained" color="primary" className={classes.themeButton} onClick={onSubmit} component="span">
                    {i18n._('Btn.Upload')}
                  </Button>
                }
              </FormControl>
            </Grid>
            <Grid item>
              {uploadData.data ? <ToastNotifier message={uploadData.data.message} type={'success'} /> : ''}
            </Grid>
            <Grid item>
              {uploadData.errorMessage ? <ToastNotifier message={uploadData.errorMessage} type={'error'} /> : ''}
            </Grid>
          </Container>
        </Paper>
      </Grid>
    </Grid>
  );
};

export default Upload;
