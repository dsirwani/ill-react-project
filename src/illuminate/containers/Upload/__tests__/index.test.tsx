import React from 'react';
import { render, fireEvent, queryByAttribute  } from "@testing-library/react";

import Upload from '../index';
import '@testing-library/jest-dom/extend-expect';

import { Provider } from 'react-redux';
import { I18nProvider, I18n } from '@lingui/react';
import { BrowserRouter } from 'react-router-dom';
import { I18nFuncContextProvider } from '../../../../hooks/useI18n';
import configureStore from 'redux-mock-store';

function createTestProps(props: any = {}) {
  return {
    // allow to override common props
    ...props,
  }
}

describe("Upload tests", () => {
  const mockStore = configureStore();
  let initialProps = {};
  let enCatalog: any;

  beforeAll(async() => {
    const locale = "en";
    enCatalog = await import(
      /* webpackMode: "lazy", webpackChunkName: "i18n-[index]" */
      `@lingui/loader!../../../../src/assets/locales/${locale}/messages`
    );
  });

  beforeEach(() => {
    initialProps = {
      uploadData: {
        loading: false,
        error: '',
        errorMessage: '',
        data: null
      },
      clientData: {
        loading: false,
        error: false,
        errorMessage: '',
        data: null,
        nextPageUrl: null,
        selectedClient: null,
        isAccountSelected: false,
        isClientSelected: false
      }
    }
  });

  const setup = (props: any) => {
    // Renders the component in memory
    const store = mockStore({ ...initialProps, ...props });
    const component = render(
      <Provider store={store}>
        <I18nProvider language={'en'} catalogs = {{'en': enCatalog}}>
          <I18n>
            {({ i18n }) => (
              <I18nFuncContextProvider value={i18n}>
                <BrowserRouter>
                  <Upload {...props} />
                </BrowserRouter>
              </I18nFuncContextProvider>
            )}
          </I18n>
        </I18nProvider>
      </Provider>
    );
    return component
  }

  /* test("Should render upload component", () => {
    const props = createTestProps()
    const tree = setup(props);
    expect(tree).toMatchSnapshot();
  }); */

  test("Should have default value for workbook drop-down", async () => {
    const props = createTestProps()
    const tree = setup(props);
    const inputDropdown = await tree.findByText('Client Workbook')
    expect(inputDropdown).not.toBe(null);
  });

  test("Should have choose file label for file input", async () => {
    const props = createTestProps()
    const tree = setup(props);
    const fileInput = await tree.findByText('Choose File')
    expect(fileInput).not.toBe(null);
  });

  test("Should have upload button for file input", async () => {
    const props = createTestProps()
    const tree = setup(props);
    const uploadButton = await tree.findByText('Btn.Upload')
    expect(uploadButton).not.toBe(null);
  });

  // test("Should test file upload with file input", async () => {
  //   const props = createTestProps()
  //   const tree = setup(props);
  //   console.log('tree: ', tree.)
  //   // const { spyOnFileChange } = jest.spyOn(tree, 'onFileChange');
  //   const getById = queryByAttribute.bind(null, 'type');
  //   const fileInput: any = getById(tree.container, 'file');
  //   // console.log('file: ', fileInput)
  //   const file = new File(['(⌐□_□)'], 'chucknorris.png', { type: 'image/png' })
  //   fireEvent.click(fileInput, { target: { files: [file] } })
  // });

});
