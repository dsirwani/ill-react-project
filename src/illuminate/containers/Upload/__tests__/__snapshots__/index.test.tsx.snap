// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload tests Should render upload component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiContainer-root MuiContainer-maxWidthSm"
            >
              <div
                class="MuiGrid-root makeStyles-nav-3 MuiGrid-item"
              >
                <a
                  href="/"
                  title="Dashboard"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root makeStyles-nav-3 MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 11H7.83l4.88-4.88c.39-.39.39-1.03 0-1.42a.9959.9959 0 00-1.41 0l-6.59 6.59c-.39.39-.39 1.02 0 1.41l6.59 6.59c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L7.83 13H19c.55 0 1-.45 1-1s-.45-1-1-1z"
                    />
                  </svg>
                </a>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <div
                  class="MuiFormControl-root makeStyles-formControl-5"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputSelect-4 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
                    data-shrink="true"
                    id="input-file-label"
                  >
                    Select File
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-inputSelect-4 MuiInputBase-formControl MuiInput-formControl"
                  >
                    <div
                      aria-haspopup="listbox"
                      aria-labelledby="input-file-label select-label"
                      class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                      id="select-label"
                      role="button"
                      tabindex="0"
                    >
                      Client Workbook
                    </div>
                    <input
                      type="hidden"
                      value="0"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSelect-icon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <div
                  class="MuiFormControl-root makeStyles-formControl-5"
                >
                  <input
                    accept=".xlsx,.xls,application/vnd.ms-excel,application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
                    class="makeStyles-inputDisplay-10"
                    id="contained-button-file"
                    type="file"
                  />
                  <label
                    class="makeStyles-fileInput-7"
                    for="contained-button-file"
                    title="Choose File"
                  >
                    <div
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-inputOverflow-8 MuiButton-outlinedPrimary"
                      role="button"
                      tabindex="0"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        <div
                          class="makeStyles-inputOverflow-8"
                        >
                          Choose File
                        </div>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </div>
                  </label>
                  <p
                    class="MuiFormHelperText-root"
                  >
                    Required
                  </p>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <div
                  class="MuiFormControl-root makeStyles-formControl-5"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-themeButton-11 MuiButton-containedPrimary"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Btn.Upload
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              />
              <div
                class="MuiGrid-root MuiGrid-item"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiContainer-root MuiContainer-maxWidthSm"
          >
            <div
              class="MuiGrid-root makeStyles-nav-3 MuiGrid-item"
            >
              <a
                href="/"
                title="Dashboard"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-nav-3 MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 11H7.83l4.88-4.88c.39-.39.39-1.03 0-1.42a.9959.9959 0 00-1.41 0l-6.59 6.59c-.39.39-.39 1.02 0 1.41l6.59 6.59c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L7.83 13H19c.55 0 1-.45 1-1s-.45-1-1-1z"
                  />
                </svg>
              </a>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiFormControl-root makeStyles-formControl-5"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputSelect-4 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
                  data-shrink="true"
                  id="input-file-label"
                >
                  Select File
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-inputSelect-4 MuiInputBase-formControl MuiInput-formControl"
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="input-file-label select-label"
                    class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                    id="select-label"
                    role="button"
                    tabindex="0"
                  >
                    Client Workbook
                  </div>
                  <input
                    type="hidden"
                    value="0"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSelect-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiFormControl-root makeStyles-formControl-5"
              >
                <input
                  accept=".xlsx,.xls,application/vnd.ms-excel,application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
                  class="makeStyles-inputDisplay-10"
                  id="contained-button-file"
                  type="file"
                />
                <label
                  class="makeStyles-fileInput-7"
                  for="contained-button-file"
                  title="Choose File"
                >
                  <div
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-inputOverflow-8 MuiButton-outlinedPrimary"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <div
                        class="makeStyles-inputOverflow-8"
                      >
                        Choose File
                      </div>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </div>
                </label>
                <p
                  class="MuiFormHelperText-root"
                >
                  Required
                </p>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiFormControl-root makeStyles-formControl-5"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-themeButton-11 MuiButton-containedPrimary"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Btn.Upload
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            />
            <div
              class="MuiGrid-root MuiGrid-item"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Upload tests should display a blank form 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiContainer-root MuiContainer-maxWidthSm"
            >
              <div
                class="MuiGrid-root makeStyles-nav-3 MuiGrid-item"
              >
                <a
                  href="/"
                  title="Dashboard"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root makeStyles-nav-3 MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 11H7.83l4.88-4.88c.39-.39.39-1.03 0-1.42a.9959.9959 0 00-1.41 0l-6.59 6.59c-.39.39-.39 1.02 0 1.41l6.59 6.59c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L7.83 13H19c.55 0 1-.45 1-1s-.45-1-1-1z"
                    />
                  </svg>
                </a>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <div
                  class="MuiFormControl-root makeStyles-formControl-5"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputSelect-4 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
                    data-shrink="true"
                    id="input-file-label"
                  >
                    Select File
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-inputSelect-4 MuiInputBase-formControl MuiInput-formControl"
                  >
                    <div
                      aria-haspopup="listbox"
                      aria-labelledby="input-file-label select-label"
                      class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                      id="select-label"
                      role="button"
                      tabindex="0"
                    >
                      Client Workbook
                    </div>
                    <input
                      type="hidden"
                      value="0"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSelect-icon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <div
                  class="MuiFormControl-root makeStyles-formControl-5"
                >
                  <input
                    accept=".xlsx,.xls,application/vnd.ms-excel,application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
                    class="makeStyles-inputDisplay-10"
                    id="contained-button-file"
                    type="file"
                  />
                  <label
                    class="makeStyles-fileInput-7"
                    for="contained-button-file"
                    title="Choose File"
                  >
                    <div
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-inputOverflow-8 MuiButton-outlinedPrimary"
                      role="button"
                      tabindex="0"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        <div
                          class="makeStyles-inputOverflow-8"
                        >
                          Choose File
                        </div>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </div>
                  </label>
                  <p
                    class="MuiFormHelperText-root"
                  >
                    Required
                  </p>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <div
                  class="MuiFormControl-root makeStyles-formControl-5"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-themeButton-11 MuiButton-containedPrimary"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Btn.Upload
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              />
              <div
                class="MuiGrid-root MuiGrid-item"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiContainer-root MuiContainer-maxWidthSm"
          >
            <div
              class="MuiGrid-root makeStyles-nav-3 MuiGrid-item"
            >
              <a
                href="/"
                title="Dashboard"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-nav-3 MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 11H7.83l4.88-4.88c.39-.39.39-1.03 0-1.42a.9959.9959 0 00-1.41 0l-6.59 6.59c-.39.39-.39 1.02 0 1.41l6.59 6.59c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L7.83 13H19c.55 0 1-.45 1-1s-.45-1-1-1z"
                  />
                </svg>
              </a>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiFormControl-root makeStyles-formControl-5"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputSelect-4 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
                  data-shrink="true"
                  id="input-file-label"
                >
                  Select File
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-inputSelect-4 MuiInputBase-formControl MuiInput-formControl"
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="input-file-label select-label"
                    class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                    id="select-label"
                    role="button"
                    tabindex="0"
                  >
                    Client Workbook
                  </div>
                  <input
                    type="hidden"
                    value="0"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSelect-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiFormControl-root makeStyles-formControl-5"
              >
                <input
                  accept=".xlsx,.xls,application/vnd.ms-excel,application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
                  class="makeStyles-inputDisplay-10"
                  id="contained-button-file"
                  type="file"
                />
                <label
                  class="makeStyles-fileInput-7"
                  for="contained-button-file"
                  title="Choose File"
                >
                  <div
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-inputOverflow-8 MuiButton-outlinedPrimary"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <div
                        class="makeStyles-inputOverflow-8"
                      >
                        Choose File
                      </div>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </div>
                </label>
                <p
                  class="MuiFormHelperText-root"
                >
                  Required
                </p>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiFormControl-root makeStyles-formControl-5"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-themeButton-11 MuiButton-containedPrimary"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Btn.Upload
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            />
            <div
              class="MuiGrid-root MuiGrid-item"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
